pyposeidon/mgmsh.py

```diff
@@ -635,34 +634,53 @@ def gmsh_execute(**kwargs):
         else:
             dim = -2
 
+        gmsh_args = kwargs.get("gmsh_args", {})
+
+        gmsh_args.update({"tol": 1e-20})
+
+        gargs = ""
+        for k, v in list(gmsh_args.items()):
+            if k[0] != "-":
+                k = "-" + k
+            gargs = gargs + " ".join([k, str(v)]) + " "
+
         # execute gmsh
-        with subprocess.Popen(
-            #            ["{} -bin -tol {} {} {}".format(bin_path, 1e-20, dim, "mymesh.geo")],
-            ["{} -tol {} {} {}".format(bin_path, 1e-20, dim, "mymesh.geo")],
+        ex = subprocess.run(
+            ["{} {} {} {}".format(bin_path, gargs, dim, "mymesh.geo")],
+            check=False,
+            capture_output=True,
+            text=True,
             cwd=calc_dir,
             shell=True,
```

Comment:
There are two options when using subprocess:
1. pass the cmd as a single string, in which case you need to also use `shell=True`
2. split the cmd using `shlex.split()` and use `shell=False` (which is the default, so you don't need to pass it).

Option 2 is considered better. Because `shell=True` can lead to potential security issues (if the cmd you execute comes from user input).